src/selectors/entities/teams.js

```diff
@@ -124,3 +133,84 @@ export const getJoinableTeams = createSelector(
         return openTeams;
     }
 );
+
+export const getJoinableTeamIds = createIdsSelector(
+    getTeams,
+    getTeamMemberships,
+    (teams, myMembers) => {
+        return Object.keys(teams).filter((id) => {
+            const team = teams[id];
+            const member = myMembers[id];
+            return team.allow_open_invite && !member;
+        });
+    }
+);
+
+export const getMySortedTeamIds = createIdsSelector(
+    getTeams,
+    getTeamMemberships,
+    (state, locale) => locale,
+    (teams, myMembers, locale) => {
+        return Object.values(teams).filter((t) => myMembers[t.id]).sort((a, b) => {
+            if (a.display_name !== b.display_name) {
+                return a.display_name.toLowerCase().localeCompare(b.display_name.toLowerCase(), locale, {numeric: true});
+            }
+
+            return a.name.toLowerCase().localeCompare(b.name.toLowerCase(), locale, {numeric: true});
+        }).map((t) => t.id);
+    }
+);
+
+export const getMyTeamsCount = createSelector(
+    getTeamMemberships,
+    (teams) => {
+        return Object.keys(teams).length;
+    }
+);
+
+// Returns the number of mentions or -1 if it has unreads
+// for every team except the current one
+export const getTeamsMentions = createSelector(
```

Comment:
Why not return the current one? It might be useful in the future if we change the UI, and the UI can handle not showing for the current channel easily enough

Reply:
Oh I see, it's not returning a map of team ids to mention counts. That name doesn't make it at all clear what this does, but I can't come up with a better one right now. Maybe something super obvious like `getChannelDrawerBadgeCount`?

I don't know how I feel about this being here then since it's pretty specific to the channel drawer button, although I guess it'll get used eventually for the mobile web view, so I guess I'm 0/5 about having it here.

Reply:
Will update the name